### ISSUE TITLE
Invalid port generated by the f:websocket element

### ISSUE THREAD

**Issue by dmatej** [payara/Payara#2170]
# Description #
----------
Some three weeks ago I implemented some JSF application with WebSockets based on one of your examples and it worked. But now it does not, browser's console reports that the connection was refused. And it is refused because of the invalid port in the url.

Log:
```
Instance Configuration
Host: localhost
HTTP Port(s): 8080 
HTTPS Port(s): 
Instance Name: payara-micro
Instance Group: no-cluster
```

`<f:websocket channel="clock" onmessage="socketListener" />`
generates
`<script id="j_idt29">jsf.push.init('j_idt29','ws://localhost:80/xxx-gui-0.0.1-SNAPSHOT/javax.faces.push/clock?244fb650-2270-4eb0-b708-69ea6d76988b','clock',null,socketListener,null,{},true);</script>`

## Expected Outcome
`<script id="j_idt29">jsf.push.init('j_idt29','ws://localhost:8080/xxx-gui-0.0.1-SNAPSHOT/javax.faces.push/clock?244fb650-2270-4eb0-b708-69ea6d76988b','clock',null,socketListener,null,{},true);</script>`

or if in appropriate part of web.xml set also 
`<transport-guarantee>CONFIDENTIAL</transport-guarantee>` and HTTPS listener on port 8443 enabled:
`<script id="j_idt29">jsf.push.init('j_idt29','wss://localhost:8443/xxx-gui-0.0.1-SNAPSHOT/javax.faces.push/clock?244fb650-2270-4eb0-b708-69ea6d76988b','clock',null,socketListener,null,{},true);</script>`

## Environment ##

- **Payara Version**: 5.xxx-SNAPSHOT (status 21. November 2017, my own compilation)
- **Edition**: Micro
- **JDK Version**: Oracle 8
- **Operating System**: Linux

**Comment by oromancer** [comment]
Fuck For Sake

**Comment by smillidge** [comment]
Do you have a simple test case we can use for debugging?

**Comment by dmatej** [comment]
It seems it is not Payara's fault, but my mojarra version compiled into my Payara build and the bug has been fixed in this commit: https://github.com/javaserverfaces/mojarra/commit/57f3e1ff948d599146f77eac4e22d012affa40ea

**Comment by michaelranaldo** [comment]
Hi @dmatej,

Thanks for updating us with your solution, glad you've gotten your project working.

**Comment by dmatej** [comment]
Fixed by #2257
